common/include/pcl/pcl_macros.h

```diff
@@ -344,3 +344,14 @@ aligned_free (void* ptr)
 #define PCL_MAKE_ALIGNED_OPERATOR_NEW \
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW \
   using _custom_allocator_type_trait = void;
+
+/**
+ * \brief Macro to add a no-op or a fallthrough attribute based on compiler feature
+ *
+ * \ingroup common
+ */
+#if __has_cpp_attribute(fallthrough) \
+  #define PCL_FALLTHROUGH [[fallthrough]];
+#else
+  #define PCL_FALLTHROUGH
```

Comment:
Either both with `;`, or both without. (There is [an opinion](https://wiki.sei.cmu.edu/confluence/display/c/PRE11-C.+Do+not+conclude+macro+definitions+with+a+semicolon) that macros should not conclude with `;`)

Reply:
`PCL_FALLTHROUGH` should not affect compilation in either mode. So adding a semicolon makes more sense than removing it from L354